apps/tactics-finder/worker.py

```
@@ -48,136 +51,178 @@ def clean_pgn(pgn: str) -> str:
 def game_to_pgn_string(game: chess.pgn.Game) -> str:
-    """Convert a chess.pgn.Game object to a PGN string."""
+    """Convert a chess.pgn.Game to PGN string."""
     exporter = chess.pgn.StringExporter(headers=True, variations=True, comments=True)
     return game.accept(exporter)
 
 
+def deterministic_puzzle_id(fen: str, moves: List[str]) -> str:
+    """Create a deterministic ID from FEN + moves (UUID-like format)."""
+    m = hashlib.sha256()
+    moves_str = ",".join(moves)
+    m.update(f"{fen}|{moves_str}".encode("utf-8"))
```

Ruff (0.14.5)

Unnecessary UTF-8 `encoding` argument to `encode`

Remove unnecessary `encoding` argument

(UP012)